src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SvDiscoveryUtils.java

```diff
@@ -78,4 +91,26 @@ private static void evaluateIntervalsAgainstTruth(final List<SVInterval> assembl
         final int nTrue = trueBreakpoints.size();
         localLogger.info("Interval false negative rate = " + falseNeg + " (" + Math.round(falseNeg*nTrue) + "/" + nTrue + ")");
     }
+
+    /**
+     * Primary reference contigs are defined as chromosomes 1-22, X, Y, M, and defined for both GRCh38 and hg19.
+     */
+    @VisibleForTesting
+    public static Set<String> getCanonicalChromosomes(final String nonCanonicalContigNamesFile, final SAMSequenceDictionary dictionary) {
+        if (nonCanonicalContigNamesFile!= null) {
+
+            try (final Stream<String> nonCanonical = Files.lines(IOUtils.getPath((nonCanonicalContigNamesFile)))) {
+                return new HashSet<>( Sets.difference(dictionary.getSequences().stream().map(SAMSequenceRecord::getSequenceName).collect(Collectors.toSet()),
+                        nonCanonical.collect(Collectors.toSet())) );
+            } catch ( final IOException ioe ) {
+                throw new UserException("Can't read nonCanonicalContigNamesFile file "+nonCanonicalContigNamesFile, ioe);
+            }
+        } else {
+            final List<String> first22ChromosomesNum = IntStream.range(0, 23).mapToObj(String::valueOf).collect(Collectors.toList());
```

Comment:
Can we not do this? It's better to require that either you have the file or you don't. If you don't have the file just treat everything as canonical. I don't want to hard code any chromosome names in the system, and I definitely don't want the tool to have any opinion on whether hg19/b37 chromosome names have "chr" in them. I know the code has been there for a while but somehow I don't remember reviewing this; maybe I just missed it in the original large pull request.

Reply:
removed